sdk/appconfiguration/azure-data-appconfiguration/README.md

```diff
@@ -167,68 +167,130 @@ ConfigurationAsyncClient client = new ConfigurationClientBuilder()
 
-Retrieve a previously stored Configuration Setting by calling getSetting.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.getSetting("some_key", "some_label");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
 ```
 
-### Update an existing Configuration Setting
+### Retrieve a configuration setting
 
-Update an existing Configuration Setting by calling setSetting.
+Retrieve a previously stored configuration setting by calling `getConfigurationSetting`.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.setSetting("some_key", "some_label", "new_value");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
+ConfigurationSetting retrievedSetting = client.getConfigurationSetting("some_key", "some_label");
+```
+For conditional request, if you want to include the ETag value of `ConfigurationSetting` in the HTTP request, 
+set the `ifChanged` parameter to true. If you decided to make a conditional request, the method could return a
+response with status code 304 (Which indicates the given configuration in the request is exactly same as in the service) if the configuration setting contains same ETag value as in the service, and response's value will be null.
+```Java
+Response<ConfigurationSetting> settingResponse = client.getConfigurationSettingWithResponse(setting, null, true, Context.NONE);
 ```
 
-### Delete a Configuration Setting
+### Update an existing configuration setting
 
-Delete an existing Configuration Setting by calling deleteSetting.
+Update an existing configuration setting by calling `setConfigurationSetting`.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.deleteSetting("some_key", "some_label");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
+ConfigurationSetting updatedSetting = client.setConfigurationSetting("some_key", "some_label", "new_value");
+```
+For conditional request, if you want to include the ETag value of `ConfigurationSetting` in the HTTP request, 
+set the `ifUnchanged` parameter to true. The configuration setting will be updated only if the ETag value matched.
+Otherwise, an exception `HttpResponseException` will be thrown. You can find what cause the exception by looking into the response's properties, `getStatusCode`, `getHeaders`, and `getRequest` if it helps.
```

Comment:
Wait, why is an exception thrown? What exceptions am I expecting?

Reply:
Oh, I was trying to explain if you can't delete the setting, then there must be an exception thrown automatically. https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationService.java#L52